infra/extfunc-lambda/lambda_function.py

```diff
@@ -0,0 +1,197 @@
+        auth = decrypt_if_encrypted(usr) + ':' + decrypt_if_encrypted(pwd)
+        req_headers['Authorization'] = b'Basic ' + b64encode(auth.encode())
+
+    # query, nextpage_path, results_path
+    req_qs = headers.get('sf-custom-querystring', '')
+    req_nextpage_path = headers.get('sf-custom-nextpage-path', '')
+    req_results_path = headers.get('sf-custom-results-path', '')
+
+    # load the data
+    body = loads(event['body'])
+    res_data = []
+    for row, *args in body.get('data'):
+        data = []
+
+        if event['path'] == '/https':
+            req_method = headers.get('sf-custom-method', 'get').format(*args).upper()
+            if 'sf-custom-json' in headers:
+                req_data = dumps(
+                    parse_header_dict(headers['sf-custom-json'].format(*args))
+                ).encode()
+            else:
+                req_data = headers.get('sf-custom-data', '').format(*args).encode()
+            next_url = req_url.format(*args)
+            next_url += ('?' + req_qs.format(*args)) if req_qs else ''
+
+            while next_url:
+                req = Request(
+                    next_url, method=req_method, headers=req_headers, data=req_data
+                )
+                links_headers = None
+                try:
+                    response = urlopen(req)
+                    links_headers = parse_header_links(
+                        ','.join(response.headers.get_all('link', []))
+                    )
+                    response_body = response.read()
+                    result = pick(req_results_path, loads(response_body))
+                except HTTPError as e:
+                    result = {'error': f'{e.status} {e.reason}'}
+                except JSONDecodeError as e:
+                    result = {
+                        'error': 'JSONDecodeError',
+                        'text': response_body.decode(),
+                    }
+
+                if req_nextpage_path and isinstance(result, list):
+                    data += result
+                    nextpage = pick(req_nextpage_path, result)
+                    next_url = f'https://{req_host}{nextpage}' if nextpage else None
+                elif links_headers and isinstance(result, list):
+                    data += result
+                    next_url = next(
+                        (l for l in links_headers if l['rel'] == 'next'), {}
+                    ).get('url')
+                else:
+                    data = result
+                    next_url = None
+
+        elif event['path'] == '/smtp':
+            header_params = {
+                k.replace('sf-custom-', '').replace('-', '_'): v.format(*args)
+                for k, v in headers.items()
+                if k.startswith('sf-custom-')
+            }
+            data = smtp(**header_params)
+
+        res_data.append([row, data])
+
+    return {'statusCode': 200, 'body': dumps({'data': res_data})}
```

Comment:
if the body variable is too long (>6MB), the code running the lambda will throw an exception. replace it here, instead.